### PR TITLE
Natron/Linux/yaml-cpp-0.6.2

### PR DESCRIPTION
**Pull request by fonkle**
Updated _libs/yaml-cpp/include/_ and _libs/yaml-cpp/src_ to the ones from verion 0.6.2 downloaded from https://github.com/jbeder/yaml-cpp/releases.
This could break with GCC < 4.0.

Note: yaml-cpp.pro was not updated.

### CODE REVIEW

**Comment by devernay** [comment]
@fonkle thanks, let's wait for the travis build.
We don't care much about gcc < 4.9 anyway, since we're using OpenMP in Natron and plugins. And C++11 will probably become a requirement very soon.
Did you get a chance to try my Dockerfile support for the SDK build?
This allows building the SDK step by step, and easy upgrading of version numbers, thanks to the docker cache. You don't have to use a VM anymore.
We now have to make the docker scripts to build Natron, and another docker script to run the tests.

File: libs/yaml-cpp/include/yaml-cpp/emitter.h [hunk 1]
```diff
@@ -122,7 +122,7 @@ class YAML_CPP_API Emitter : private noncopyable {
   bool CanEmitNewline() const;
 
  private:
-  std::auto_ptr<EmitterState> m_pState;
```
Comment (devernay) [inline libs/yaml-cpp/include/yaml-cpp/emitter.h]: this means that C++11 is required

**Comment by devernay** [comment]
ok so it's either linking with the wrong yaml (it should be linkng with yaml-cpp-natron), or it doesn't compile yaml-cpp with -DYAML=YAML_NATRON.
just modify the build scripts to add "V=1" to the command line that calls make for Natron

**Comment by devernay** [comment]
And please add a C++11 compiler to the requirements :
- minimum GCC version is 4.8.1, recommended is 6.1 or more
- minimum clang version is 3.3, recommended is 6.0 or more
- minimum Visual Studio version is VS2015

**Comment by devernay** [comment]
to generate the dockerfile:
env GEN_DOCKERFILE=1 include/scripts/build-Linux-sdk.sh > path_to/Dockerfile
then go to the Dockerfile dir and execcute
```
docker build -t natronsdk:latest .
```

**Comment by fonkle** [comment]
Oke, I will try to get to grips with Docker.

**Comment by devernay** [comment]
easy (since my last push):
- install docker
- execute tools/docker/natron-sdk/build.sh to build the natron-sdk docker image
- execute `docker run -it natron-sdk:latest sh` to get a shell inside the natron-sdk image

**Comment by fonkle** [comment]
I get it and got it running, building the SDK. Nice.

If it succeeds, will the launchBuildMain.sh script be used for building Natron in the same container?

**Comment by fonkle** [comment]
The build works (Pyside had an issue with not finding _FindShiboken.cmake_ but the image (3.11 GByte) called got created.

What's your plan with building Natron itself? Do you want to write a Dockerfile for that as well?

**Comment by devernay** [comment]
I planned to just run the build script over the natron-sdk image, and copy the result to the host filesystem. I'm still working on the natron-sdk image.

**Comment by fonkle** [comment]
Awesome! I will patiently wait. Looking forward to testing the build again :).

**Comment by fonkle** [comment]
FYI, the updated version of the SDK (RB-2.3-sdk-upgrade on 25-05-2019) builds fine using docker here.

**Comment by devernay** [comment]
great! I still haven't got a chance to complete the build! I'll push it to dockerhub as soon as I have it (not before tuesday, since I'm away from home)

**Comment by fonkle** [comment]
Good work, almost everything is updated :).

So would you then clone the Natron repo inside the container (install OCIO-configs, config.pri and env variables) and:
1. run qmake -r and make?
2. run the scripts (build-Natron.sh and build-Linux-installer.sh) in tools/jenkins?
3. run the launchBuildMain.sh script in tools/jenkins?

What is the best way to build a development version?

**Comment by devernay** [comment]
I'm not yet there, but I would launch a bash shell in docker, while sharing the directory where I compile everything, so that things stay where they are.
something like `docker run -v /host/directory:/home natron-sdk:latest bash`
then do everything inside home.
launchBuildMain should work to build a snapshot and test it, but you should first try to launch it from that bash shell.

If you want to try the "manual" way, I usually do something like that to setup the SDK environment (you may want the change the directories where the Natron sources are:
```
SDK=/opt/Natron-sdk
GCC="$SDK/gcc"
QTDIR="$SDK/qt4"
FFMPEG="$SDK/ffmpeg-gpl2"
LIBRAW="$SDK/libraw-gpl2"
PATH="$SDK/bin:$QTDIR/bin:$GCC/bin:$FFMPEG/bin:$LIBRAW_PATH:$PATH"
LIBRARY_PATH="$SDK/lib:$QTDIR/lib:$GCC/lib64:$GCC/lib:$FFMPEG/lib:$LIBRAW/lib"
LD_LIBRARY_PATH="$SDK/lib:$QTDIR/lib:$GCC/lib64:$GCC/lib:$FFMPEG/lib:$LIBRAW/lib"
LD_RUN_PATH="$SDK/lib:$QTDIR/lib:$GCC/lib:$FFMPEG/lib:$LIBRAW/lib"
CPATH="$SDK/include:$QTDIR/include:$GCC/include:$FFMPEG/include:$LIBRAW/include"
export PKG_CONFIG_PATH="$SDK/lib/pkgconfig:$SDK/osmesa/lib/pkgconfig:$QTDIR/lib/pkgconfig:$GCC/lib/pkgconfig:$FFMPEG/lib/pkgconfig:$LIBRAW/lib/pkgconfig"
export QTDIR LD_LIBRARY_PATH LD_RUN_PATH PKG_CONFIG_PATH CPATH
NATRON="$HOME/Development/Natron-gforge"
NATRON_PLUGIN_PATH="$NATRON/Natron-gforge/Gui/Resources/PyPlugs"
OFX_PLUGIN_PATH="$HOME/Development/OFX/Plugins"
export NATRON_PLUGIN_PATH OFX_PLUGIN_PATH
export LANG="C"
export OCIO="$HOME/Development/Natron-gforge/OpenColorIO-Configs/blender/config.ocio"
echo "compile natron with:"
echo "qmake CONFIG+=debug CONFIG+=enable-cairo CONFIG+=enable-osmesa CONFIG+=openmp LLVM_PATH=$SDK/llvm OSMESA_PATH=$SDK/osmesa -r"
```

The easiest way to *debug and develop* Natron, however, is to install the SDK inside the linux distrib. the build-Linux-sdk.sh script should work on any distrib, just make sure that /opt/Natron-sdk is user-writable.

**Pull request closed without merging**